applications: nrf_desktop: Simplify HIDS configuration (#10047)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/hids.rst

1. Complete the basic Bluetooth configuration, as described in :ref:`nrf_desktop_bluetooth_guide`.
   Make sure that both :ref:`CONFIG_DESKTOP_ROLE_HID_PERIPHERAL <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_BT_PERIPHERAL <config_desktop_app_options>` options are enabled.
   The HID Service application module is enabled by the :ref:`CONFIG_DESKTOP_HIDS_ENABLE <config_desktop_app_options>` option which is selected by :ref:`CONFIG_DESKTOP_BT_PERIPHERAL <config_desktop_app_options>` together with other GATT Services that are required for a HID device.
#. The :ref:`CONFIG_DESKTOP_HIDS_ENABLE <config_desktop_app_options>` option selects the following Kconfig options:

>    * :kconfig:option:`CONFIG_BT_HIDS` (to automatically enable the :ref:`hids_readme`)

Reviewer: ```suggestion
   * The :kconfig:option:`CONFIG_BT_HIDS` option that automatically enables the :ref:`hids_readme`.
```

   * :kconfig:option:`CONFIG_BT_CONN_CTX` (to automatically enable the ref:`bt_conn_ctx_readme`, which is required by the |GATT_HID|).

#. The nRF Desktop application modifies defaults of Kconfig option values defined by the :ref:`hids_readme` to tailor the default configuration to application needs.
   The configuration is tailored for either nRF Desktop mouse (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_MOUSE <config_desktop_app_options>`) or nRF Desktop keyboard (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_KEYBOARD <config_desktop_app_options>`).

>    See the :file:`src/modules/Kconfig.hids` file for details.

Reviewer: ```suggestion
   For more details, see the :file:`src/modules/Kconfig.hids` file.
```

   .. tip::
     If the HID report configuration is identical to the default configuration of either nRF Desktop mouse or keyboard, you do not need to modify the |GATT_HID| configuration.

>      Otherwise, see documentation of the ref:`hids_readme` for configuration details.

Reviewer: ```suggestion
     Otherwise, see :ref:`hids_readme` documentation for configuration details.
```

.. note::
   The nRF Desktop centrals perform the GATT service discovery and reenable the HID notifications on every reconnection.
   A HID report that is received before the subscription is reenabled will be dropped before it reaches the application.

>    The :ref:`CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY <config_desktop_app_options>` is set to 500 ms for nRF Desktop keyboards (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_KEYBOARD <config_desktop_app_options>`) to make sure that the input will not be lost on reconnection with the nRF Desktop dongle.

Reviewer: ```suggestion
   The :ref:`CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY <config_desktop_app_options>` option is set to 500 ms for nRF Desktop keyboards (:ref:`CONFIG_DESKTOP_PERIPHERAL_TYPE_KEYBOARD <config_desktop_app_options>`) to make sure that the input is not lost on reconnection with the nRF Desktop dongle.
```
